src/main/java/org/javarosa/core/model/instance/ExternalDataInstanceSource.java

```diff
@@ -83,52 +105,60 @@ public void init(TreeElement root) {
     }
 
     public void remoteInit(RemoteInstanceFetcher remoteInstanceFetcher) throws RemoteInstanceFetcher.RemoteInstanceException {
-        try {
-            init(remoteInstanceFetcher.getExternalRoot(instanceId, this));
-            root.setInstanceName(instanceId);
-            root.setParent(new InstanceBase(instanceId));
-        } catch (IOException e) {
-            throw new RemoteInstanceFetcher.RemoteInstanceException(
-                    "Could not retrieve data for remote instance " + instanceId + ". Please try opening the form again.", e);
-        } catch (XmlPullParserException | UnfullfilledRequirementsException | InvalidStructureException e) {
-            throw new RemoteInstanceFetcher.RemoteInstanceException(
-                    "Invalid data retrieved from remote instance " + instanceId+ ". If the error persists please contact your help desk.", e);
-        }
+        String instanceId = getInstanceId();
+        init(remoteInstanceFetcher.getExternalRoot(instanceId, this));
+        root.setInstanceName(instanceId);
+        root.setParent(new InstanceBase(instanceId));
+    }
+
+    public void setupNewCopy(ExternalDataInstance instance) {
+        instance.copyFromSource(this);
     }
 
     @Override
     public void readExternal(DataInputStream in, PrototypeFactory pf) throws IOException, DeserializationException {
-        sourceUri = ExtUtil.readString(in);
         instanceId = ExtUtil.readString(in);
         useCaseTemplate = ExtUtil.readBool(in);
+        sourceUri = ExtUtil.nullIfEmpty(ExtUtil.readString(in));
         requestData = (Multimap<String, String>)ExtUtil.read(in, new ExtWrapMultiMap(String.class), pf);
+        Object nullableRefId = ExtUtil.read(in, new ExtWrapNullable(String.class), pf);
```

Comment:
These serialisation changes are now resulting in breaking the deserialisation for a serialised form def (https://github.com/dimagi/formplayer/pull/1126/files#r868345569) and we will need to implement a fallback mechanism against such errors before deploying it to production.